PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -138,16 +138,30 @@ def nanoAOD_addBoostedTauIds(process, idsToRun=[]):
 
     return process
 
-def nanoAOD_addPNetToTaus(process, addPNetInfo=False, runPNetCHSAK4=False):
-    if addPNetInfo:
+def nanoAOD_addUTagToTaus(process, addUTagInfo=False, runUTagAK4=False,
+                          usePUPPIjets=False):
+    if addUTagInfo:
         originalTauName = process.finalTaus.src.value()
-        updatedTauName = originalTauName+'WithPNet'
-        jetCollection = "updatedJets"
-        process.load('RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff')
-        pnetTagName = "pfParticleNetFromMiniAODAK4CHSCentralJetTag"
-        pnetDiscriminators = [];
-        for tag in getattr(process,pnetTagName+"s").flav_names.value():
-            pnetDiscriminators.append(pnetTagName+"s:"+tag)
+        
+        if usePUPPIjets: # option to use PUPPI jets   
+            jetCollection = "updatedJetsPuppi"
+            TagName = "pfUnifiedParticleTransformerAK4JetTag"
+            tag_prefix = "byUTagPUPPI"
+            updatedTauName = originalTauName+'WithUTagPUPPI'
+            # Unified ParT Tagger used for PUPPI jets
+            from RecoBTag.ONNXRuntime.pfUnifiedParticleTransformerAK4JetTags_cfi import pfUnifiedParticleTransformerAK4JetTags
+            process.pfUnifiedParticleTransformerAK4JetTags = pfUnifiedParticleTransformerAK4JetTags.clone()
```

Comment:
```suggestion
            from RecoBTag.ONNXRuntime.pfUnifiedParticleTransformerAK4_cff import pfUnifiedParticleTransformerAK4JetTags
            Discriminators = pfUnifiedParticleTransformerAK4JetTags.flav_names.value()
```

- Please avoid attaching `pfUnifiedParticleTransformerAK4JetTags` to `process` as you do not intend to run it as a module.

Reply:
Sure, will implement in an alternative way.